apis/config/v1beta1/configuration_types.go

```diff
@@ -352,3 +355,27 @@ type ClusterQueueVisibility struct {
 	// Defaults to 10.
 	MaxCount int32 `json:"maxCount,omitempty"`
 }
+
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
+)
+
+type FairSharing struct {
+	// enable indicates whether to enable fair sharing for all cohorts.
+	// Defaults to false.
+	Enable bool `json:"enable"`
+
+	// preemptionStrategies indicates which constraints should a preemption satisfy.
+	// The preemption algorithm will only use the next strategy in the list if the
+	// incoming workload (preemptor) doesn't fit after using the previous strategies.
+	// Possible values are:
+	// - LessThanOrEqualToFinalShare: Only preempt if the share of the preemptor CQ
+	//   will be less than or equal to the share of the preemptee CQ after the preemption occurs.
+	// - LessThanInitialShare: Only preempt if the share of the preemptor CQ will be strictly less
+	//   than the share of the preemptee CQ before the preemption.
+	// The default strategy is ["LessThanOrEqualToFinalShare"].
+	PreemptionStrategies []PreemptionStrategy `json:"preemptionStrategies,omitempty"`
```

Comment:
I don't have a strong view here, but I would suggest to consider dropping this configuration, unless we have some evidence users will need it, because:
1. the validation and parsing adds a substantial amount of code, asking for coverage. In particular we don't have integration tests for only one of the rules enabled
2. the strategies are quite complex, and I don't think users would like to configure them on their own (unless we have some evidence for that)
3. I'm not sure the list of strategies is the correct abstraction for the API. Do we foresee more strategies in this form? Maybe dedicated booleans is enough? Say "lessThanOrEqualToFinalShare: true/false`, etc. 

Also, the API is not discussed in KEP, so please update the KEP API accordingly if we go this path.

Reply:
I do have evidence of a need for a configuration like `["LessThanInitialShare"]`, which was not originally envisioned in the KEP.

Then the question is to @mwielgus and @tenzen-y if you would rather just keep this only configuration, or you still want to keep the other two possibilities (`["LessThanOrEqualToFinalShare"]` and ["LessThanOrEqualToFinalShare", "LessThanInitialShare"]`).

> In particular we don't have integration tests for only one of the rules enabled

That is true, but the different algorithms don't change the mechanics of actually doing the preemptions. Integration tests just make sure that multiple preemptions can occur, regardless of the algorithm.

> Do we foresee more strategies in this form?

I'm not certain of that. I think we might just have the two strategies. But again, this leads to three different configurations, and an array seems to be the easiest way to communicate the semantics.

Reply:
I had a discussion with @mwielgus and @mimowo. We brainstormed a lot of names and a simplified APIs, but ended up using terms that are not immediately obvious, such as `balance`, `minimalShare`, `soft`.

So, we decided to keep the current proposal, which is more mathematical, assuming that whoever is looking to change the default needs to read the documentation and understand some of the basics of the algorithm.

Reply:
SGTM

@alculquicondor @mimowo @mwielgus I'm sure that flexibility and understandability are tradeoffs. I'm curious about whether we can provide the `balance`, `minimalShare`, and `soft` via the `kueuectl` since I can imagine that the user of `kueuectl` would prefer to avoid the deep dive into the Kueue.

Reply:
`kueuectl` is not meant to change Kueue configuration. I will have to do a good job via documentation, instead.

Reply:
SGTM